components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/internal/TemplateManagerComponent.java

```diff
@@ -100,4 +129,124 @@ protected void unsetConfigurationManager(ConfigurationManager configurationManag
+                                    getFileBasedTemplates().put(templateObj.getString(TemplateMgtConstants.ID),
+                                            template);
+                                } catch (IOException e) {
+                                    log.error("Error while reading  templates.", e);
+                                }
+                            });
+                } catch (IOException e) {
+                    log.error("Error while reading templates.", e);
+                }
+            }
+        }
+    }
+
+    private Map<String, String> extractApplicationSpecificProperties(JSONObject templateObj) {
+
+        Map<String, String> properties = new HashMap<>();
+        if (StringUtils.isNotEmpty(templateObj.getString(TemplateMgtConstants.AUTHENTICATION_PROTOCOL))) {
+            properties.put(TemplateMgtConstants.AUTHENTICATION_PROTOCOL, templateObj.getString(TemplateMgtConstants
+                    .AUTHENTICATION_PROTOCOL));
+        }
+        if (templateObj.getJSONArray(TemplateMgtConstants.TYPES) != null) {
+            JSONArray typesJSONArray = templateObj.getJSONArray(TemplateMgtConstants.TYPES);
+            List<String> types = new ArrayList<>();
+            for (int i = 0; i < typesJSONArray.length(); i++) {
+                types.add(typesJSONArray.getString(i));
+            }
+            properties.put(TemplateMgtConstants.TYPES, String.join(",", types));
+        }
+        if (StringUtils.isNotEmpty(templateObj.getString(TemplateMgtConstants.CATEGORY))) {
+            properties.put(TemplateMgtConstants.CATEGORY, templateObj.getString(TemplateMgtConstants.CATEGORY));
+        }
+        if (StringUtils.isNotEmpty(String.valueOf(templateObj.getInt(TemplateMgtConstants.DISPLAY_ORDER)))) {
+            properties.put(TemplateMgtConstants.DISPLAY_ORDER, Integer.toString(templateObj.getInt
+                    (TemplateMgtConstants.DISPLAY_ORDER)));
+        }
+        return properties;
+    }
+
+    private Map<String, String> extractIDPSpecificProperties(JSONObject templateObj) {
+
+        Map<String, String> properties = new HashMap<>();
+        if (StringUtils.isNotEmpty(templateObj.getString(TemplateMgtConstants.PROP_CATEGORY))) {
+            properties.put(TemplateMgtConstants.PROP_CATEGORY, templateObj.getString(TemplateMgtConstants
+                    .PROP_CATEGORY));
+        }
+        if (StringUtils.isNotEmpty(String.valueOf(templateObj.getInt(TemplateMgtConstants.PROP_DISPLAY_ORDER)))) {
+            properties.put(TemplateMgtConstants.PROP_DISPLAY_ORDER, Integer.toString(templateObj.getInt
+                    (TemplateMgtConstants.PROP_DISPLAY_ORDER)));
+        }
+        if (templateObj.getJSONArray(TemplateMgtConstants.PROP_SERVICES) != null) {
+            JSONArray typesJSONArray = templateObj.getJSONArray(TemplateMgtConstants.PROP_SERVICES);
+            List<String> services = new ArrayList<>();
+            for (int i = 0; i < typesJSONArray.length(); i++) {
```

Comment:
check whether typesJSONArray is not  null before accessing ypesJSONArray.length()